llm_on_ray/inference/predictor_deployment.py

```diff
@@ -93,145 +104,253 @@ async def consume_streamer_async(self, streamer: TextIteratorStreamer):
                 # back to the event loop so other coroutines can run.
                 await asyncio.sleep(0.001)
 
-    async def __call__(self, http_request: Request) -> Union[StreamingResponse, JSONResponse, str]:
-        json_request: Dict[str, Any] = await http_request.json()
-        prompts = []
-        text = json_request["text"]
-        config = json_request["config"] if "config" in json_request else {}
-        streaming_response = json_request["stream"]
-        if isinstance(text, list):
-            prompt_format = get_prompt_format(text)
-            if prompt_format == PromptFormat.CHAT_FORMAT:
-                if self.process_tool is not None:
-                    prompt = self.process_tool.get_prompt(text)
-                    prompts.append(prompt)
-                else:
-                    prompts.extend(text)
-            elif prompt_format == PromptFormat.PROMPTS_FORMAT:
-                if streaming_response:
-                    return JSONResponse(
-                        status_code=400,
-                        content="Multiple prompts are not supported when streaming response is enabled.",
-                    )
-                prompts.extend(text)
-            else:
-                return JSONResponse(status_code=400, content="Invalid prompt format.")
-        else:
-            prompts.append(text)
-
-        if not streaming_response:
-            if self.use_vllm:
-                return await self.predictor.generate_async(prompts, **config)
+    # Handle streaming, only support single prompt
+    async def handle_streaming(self, prompt: str, config: Dict[str, Any]):
+        if isinstance(prompt, list):
+            error_message = (
+                "Streaming response is not supported when multiple prompts are provided."
+            )
+            if not self.use_openai:
+                yield JSONResponse(
+                    status_code=400,
+                    content=error_message,
+                )
             else:
-                return self.predictor.generate(prompts, **config)
-
+                yield ModelResponse(
+                    error=ErrorResponse(
+                        message=error_message,
+                        code=400,
+                        internal_message=error_message,
+                        type="InternalServerError",
+                    )
+                )
         if self.use_deepspeed:
-            self.predictor.streaming_generate(prompts, self.streamer, **config)
-            return StreamingResponse(
-                self.consume_streamer(), status_code=200, media_type="text/plain"
-            )
+            self.predictor.streaming_generate(prompt, self.streamer, **config)
+            if not self.use_openai:
+                yield StreamingResponse(
```

Comment:
I feel this file is too complex, too many if else. Can we further simplify this? For example, if we are returning ModelResponse for openai, can we return SimpleModelResponse for simple serving server, but the logic to create the response can be the same. So you can create a method to generate the response without if else every where? Also, can we refactor for the predictor classes so we don't need to check which predictor we are using? For example, call the same interface for streaming.

Reply:
The primary purpose of the PR is to add dynamic batching using the original code structure. If have concerns here, I would suggest open another PR to do the refactoring and leave this PR to add dynamic batching later.